Add a basic description of parameters when using the custom connection type in database replication (#8057)
mendix/docs: content/en/docs/appstore/use-content/modules/database-replication.md

You can create new database connections from the **Databases** page. Each connection has a few basic properties. After creating a database connection, you should synchronize the table information using the button on the overview. This makes sure the connection information is correct, and that the replication module has all the table information from your database.

> In this page you must select the database type before filling in the connection details. Available database parameters vary per database type.

Reviewer: Proofread
```suggestion
On this page you must select the database type before filling in the connection details. Available database parameters vary depending on database type.
```

> #### 2.1.1 Common Parameters

Reviewer 2: I suspect that your goal is not to list available databases, but I would still create a short comment for every field.
Especially the following ones:
- Service name and SID — I'd mention that these are only for Oracle and that one excludes the other
- Use Integrated Security — this one is only for SQL Server

Reviewer: I agree, if these are common parameters, then it shouldn't depend on the database type as to which ones become available (see below).
If some of these are specific to specific cases (even if that case is "not Oracle" then I think it should be in a separate section).

Author: I tried adding a short description but all I got was noise. They are either self explanatory while reading the list or only relevant for the database types they appear on (and in which case it should be clear to the user what it is).

The only possible exception is `Use Integrated Security`, which seems to also be available for Informix as well.

Reviewer: I agree there is no point in repeating the in-app explanation. But perhaps a short note that says "Oracle databases only" or something like that?

Depending on which database type is selected, some of the following parameters become available:
- Host
- Port
- Username
- Password
- Database name
- Instance name
- Service name
- SID
- Use Integrated Security

#### 2.1.2 Custom Connection

If your database type is not listed, it may still be possible to connect using a JDBC compatible driver and the **Custom** connection option. You need to obtain the drivers and place them inside the `userlib` directory and then configure the connection parameters:

> - Username and Password: The credentials given to the driver to connect to the database server;

Reviewer: Should we mention here not to hard code them in the app? Here is a good link: https://docs.mendix.com/refguide/dev-best-practices/#25-passwords-and-other-secrets (although I'll need to set up a permanent anchor).

Author: There is no hard coding here. This page is filled after the app is deployed and the credentials are stored in the database (which is just as bad, IMO, but that's how it currently is implemented).

Reviewer: OK - then the security is managed for them. I'll resolve this one.

> - Connection string: A fully formed JDBC URL string (e.g. `jdbc:postgresql://localhost:5432/database` for a Postgres databas);

Reviewer 2: *database

Reviewer: ```suggestion
- Connection string: A fully formed JDBC URL string (e.g. `jdbc:postgresql://localhost:5432/database` for a PostgreSQL database);
```

> - Open/Close escape character: Characters used to quote table and column names (e.g. `"` in Postgres or `[` and `]` in SQL Server);

Reviewer: I think the name of the product in PostgreSQL, so we should say:
```suggestion
- Open/Close escape character: Characters used to quote table and column names (e.g. `"` in PostgreSQL or `[` and `]` in SQL Server);
```

> - Driver class: The full name of the driver to be loaded (e.g. `org.postgresql.Driver` for Postgres);

Reviewer: ```suggestion
- Driver class: The full name of the driver to be loaded (e.g. `org.postgresql.Driver` for PostgreSQL);
```

- Escape column names: Indicates if column names must be quoted using the escape characters when querying the database;
- Escape table names: Indicates if table names must be quoted using the escape characters when querying the database;
- Escape the schema and table name separately: Indicates if we should quote schema and table names together (e.g. `"public_schema.table_name"`) or separately (e.g. `"public_schema"."table_name"`);
- Allow AS token for tables: Indicates if `AS` is necessary when aliasing table names (e.g. `SELECT * FROM table AS A` vs `SELECT * FROM table A`);

> - Close connection after query: Indicates if the connection to the database must be closed after every query.

Reviewer 2: In the list in 2.1.1 there is nothing in the end of the string. In this list you use `;` and `.` — perhaps those can be dropped as well

Reviewer: ➕

Author: If anything, I'd add a `;` and `.` to the other list (following what seems to be the AP Stylebook). I couldn't find any mention of which style to use in our documentation, though.

Author: Actually, it does not look like it's AP's, but someone else's.

Reviewer: There is something in Confluence - but it isn't clear. I'll update it.

Unordered lists should either not have any punctuation at the end, or should have a full stop if each item completes a sentence which introduces the list.

In this case, I would probably put a full stop at the end (although this is one thing which I am never 100% sure about😀).

Reviewer 2: Usually a label should have a full stop at the end if there are multiple sentences. If it has one sentence, no punctuation is required. For lists, I tend to follow the same rule.

Reviewer: Oh, I've found the styling for bullets - I'll Slack you a link.

Author: I replaced them with periods in this list as there is a full description after each entry. The other lists I left period-less.

* Username and Password: The credentials given to the driver to connect to the database server.
* Connection string: A fully formed JDBC URL string (for example, `jdbc:postgresql://localhost:5432/database` for a PostgreSQL database).
* Open/Close escape character: Characters used to quote table and column names (for example, `"` in PostgreSQL or `[` and `]` in SQL Server).
* Driver class: The full name of the driver to be loaded (for example, `org.postgresql.Driver` for PostgreSQL).
* Escape column names: Indicates if column names must be quoted using the escape characters when querying the database.
* Escape table names: Indicates if table names must be quoted using the escape characters when querying the database.
* Escape the schema and table name separately: Indicates if we should quote schema and table names together (for example, `"public_schema.table_name"`) or separately (for example, `"public_schema"."table_name"`).
* Allow AS token for tables: Indicates if `AS` is necessary when aliasing table names (for example, `SELECT * FROM table AS A` vs `SELECT * FROM table A`).

> * Close connection after query: Indicates if the connection to the database must be closed after every query.

Reviewer: Should we indicate that these are Booleans. We imply that the other parameters are strings by indicating what goes into them.

Author: The boolean options are displayed as checkboxes, so I don't think (within their context) that this clarification is necessary.

Reviewer: OK - we've discussed. I'll resolve this.
